logstash-core/lib/logstash/instrument/collector.rb

```diff
@@ -37,7 +37,22 @@ def push(namespaces_path, key, type, *metric_type_params)
           LogStash::Instrument::MetricType.create(type, namespaces_path, key)
         end
 
-        metric.execute(*metric_type_params)
+        unless metric_type_params.size.eql? 2
+          logger.error("Collector: Cannot record metric. Unexpected metric params: " + metric_type_params.to_s)
+          return
+        end
+
+        method = metric_type_params[0]
```

Comment:
@jakelandis this variable seems to only be used in the `else` branch, I think we can inline it there?